ru/learn/variables/array.md

```diff
@@ -0,0 +1,847 @@
+my_array = array.array('i', [1, 2, 3, 4, 5])
+```
+
+2. Объявление списка вместо массива:
+
+```python
+# Create a list
+my_list = [1, 2, 3, 4, 5]
+```
+
+Чтобы создать пустой массив, вы можете следовать упомянутым выше подходам. Далее мы рассмотрим определение массива размером `n`.
+
+### Массив Размера `n`
+
+Для создания массива определенного размера в Python вы можете использовать различные методы, включая использование генератора списка или NumPy. Вот несколько примеров объявления массивов:
+
+Использование генератора списка:
+
+```python
+n = 5
+my_array = [0] * n
+print(my_array) # Output: [0, 0, 0, 0, 0]
+```
+
+Использование NumPy:
+
+```python
+import numpy as np
+
+n = 5
+my_array = np.zeros(n)
+print(my_array) # Output: [0. 0. 0. 0. 0.]
+```
+
+### Массив, Генерируемый Случайным Образом
+
+Для генерации случайного массива в Python вы можете использовать `модуль random` из стандартной библиотеки Python или библиотеку numpy. Вот примеры, использующие оба подхода:
+
+Использование модуля random:
+
+```python
+import random
+
+# Generate a random array of size n
+n = 5
+random_array = [random.randint(0, 100) for _ in range(n)]
+
+print(random_array)  # Output: [47, 92, 32, 12, 80]
+```
+
+Использование библиотеки numpy:
+
+```python
+import numpy as np
+
+# Generate a random array of size n
+n = 5
+random_array = np.random.randint(0, 100, size=n)
+
+print(random_array)  # Output: [92 76 32 48 11]
+```
+
+Оба подхода позволяют вам генерировать массивы случайных целых чисел. Настройте параметры (`a`, `b` и `size`) в зависимости от ваших конкретных требований, чтобы контролировать диапазон и размер случайного массива.
+
+### 2D Массив в Python
+
+Вот пример того, как инициализировать многомерный массив в Python с использованием функции `np.array()`:
+
+```python
+import numpy as np
+
+# Create a 2-dimensional array
+arr2 = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+print(arr2)
+# Output:
+# [[1 2 3]
+#  [4 5 6]
+#  [7 8 9]]
+```
+
+Вы также можете создать двумерный массив, используя список списков, где каждый внутренний список представляет собой строку. Вот пример создания и инициализации двумерного массива с использованием вложенных списков:
+
+```python
+# Create a 2D array with 3 rows and 4 columns
+array_2d = [[0, 1, 2, 3],
+            [4, 5, 6, 7],
+            [8, 9, 10, 11]]
+
+# Accessing elements
+print(array_2d[0][2])  # Output: 2 (element at row 0, column 2)
+print(array_2d[1][3])  # Output: 7 (element at row 1, column 3)
+
+# Modifying elements
+array_2d[2][1] = 99  # Changing element at row 2, column 1 to 99
+print(array_2d[2][1])  # Output: 99
+
+# Iterating through the array
+for row in array_2d:
+    for element in row:
+        print(element, end=' ')
+    print()
+```
+
+### Как Создать Массив NumPy в Python
+
+Чтобы создать [массив NumPy](https://numpy.org/doc/stable/reference/generated/numpy.array.html) в Python, вы можете использовать функцию `numpy.array()`. Вот пример инициализации массива np:
```

Comment:
The `np` can be highlighted as a code. If you see such mistakes - feel free to fix them in both language versions.